src/api.yaml

```diff
@@ -0,0 +1,1285 @@
+        the version without `/filename` for a JSON-returning version."
+      parameters:
+        - $ref: "#/components/parameters/pathRoomToken"
+        - $ref: "#/components/parameters/pathFileId"
+        - name: filename
+          in: path
+          required: true
+          description: >
+            Filename if known by the requesting client, and empty otherwise.  The value of this
+            parameter is ignored by the server itself: it is included to differentiate this request
+            from the JSON version, and so that clients may include a filename in the request URL for
+            contexts where that is useful.
+          schema:
+            type: integer
+            format: int64
+      responses:
+        200:
+          description: successful operation; returns the file, in raw bytes.
+          headers:
+            Content-Length:
+              description: The size of the file.
+              schema:
+                type: integer
+                format: int64
+              example: 12345
+            Date:
+              description: The HTTP timestamp at which the file was uploaded.
+              schema:
+                type: string
+                example: "Thu, 7 Oct 2021 00:42:00 GMT"
+            Expires:
+              description: >
+                The HTTP timestamp at which the file is scheduled to expire.  This header is omitted
+                if the attachment is non-expiring (e.g. for attachments in a pinned message).
+              schema:
+                type: string
+                example: "Fri, 22 Oct 2021 00:42:42 GMT"
+          content:
+            application/octet-stream:
+              schema:
+                type: string
+                format: binary
+        403:
+          $ref: "#/paths/~1room~1%7BroomToken%7D~1file~1%7BfileId%7D/get/responses/403"
+        404:
+          $ref: "#/paths/~1room~1%7BroomToken%7D~1file~1%7BfileId%7D/get/responses/403"
+
+                    
+  #/user/{sessionId}/
+  /user/{sessionId}/ban:
+    post:
+      tags: [Users]
+      summary: Bans or unbans a user.
+      description: >
+        Applies or removes a ban of a user from specific rooms, or from the server globally.
+        
+        
+        Note that the given session ID does not have to exist: it is possible to preemptively ban
+        users who have never visited the server or room(s).
+        
+        
+        The user's messages are not deleted by this request.  In order to ban and delete all messages
+        use the `/sequence` endpoint to bundle a `/user/.../ban` with a `/user/.../deleteMessages`
+        request.
+      parameters:
+      - $ref: "#/components/parameters/pathSessionId"
+      requestBody:
+        description: "Details of the ban to apply. To unban a user, specify a negative timeout."
+        required: true
+        content:
+          application/json:
```

Comment:
we should require a ban reason in this structure, for no reason set it to null, otherwise it contains why they were banned.

also we should consider adding ban appeal endpoints.